doc-tools/missing-doclet/src/main/java/org/opensearch/missingdoclet/MissingDoclet.java

```diff
@@ -291,6 +293,22 @@ private void checkComment(Element element) {
         if (ignored.contains(element.toString())) {
             return;
         }
+        // Ignore classes annotated with @Generated and all enclosed elements in them.
+        if (element.getAnnotation(javax.annotation.Generated.class) != null) {
```

Comment:
So there is a way to make these checks **without** requiring `javax.annotation.Generated` to be on classpath of the `MissingDoc` (basically only `server` would need it). Instead of using `element.getAnnotation()` directly (it needs class), we could use `element.getAnnotationMirrors()` and compare class names as strings:

```
    // Ignore classes annotated with @Generated and all enclosed elements in them.
    private boolean isGenerated(Element element) {
        return element
             .getAnnotationMirrors()
             .stream()
             .anyMatch(m -> m
                     .getAnnotationType()
                     .toString() /* ClassSymbol.toString() returns class name */
                     .equalsIgnoreCase("javax.annotation.Generated")); 
    }

```
The `toString()` may look a bit shaky - but the API for ClassSymbol (internal) does return class full name in the implementation.  The check becomes:

```
        if (isGenerated(element)) {
            return;
        }
```

Reply:
It doesn't work so I'll need to debug what's going on.  I do like this approach as I do not want to introduce the jboss dependency into this module.

Maybe we should add some unit tests to 'missing-doclet'?  I tried pulling in ":test:framework" but to no avail.  It is another huge dependency chain to bring along.  Can we have some simple JUnit tests in doc-tools that do not extend OpenSearchTestCase?  What do you think?

Reply:
@austintlee would you mind to expand "It doesn't work", I tested things locally before, do you prefer me to push changes into your pull request? Thank you.

Reply:
"gradlew javadoc precommit" with the above change failed with:

```
error: org.opensearch.extensions.proto (package): javadocs are missing
```

In your local testing, are you using my code that ignores packages?

Reply:
:D I know why, I run into this issue when translating the condition below (should be ` (!isGenerated(e)):`): 

```
Optional<?> elm = enclosedElements.stream().findFirst().filter(e -> ((e.getKind() != ElementKind.CLASS) || (!isGenerated(e))));
```

Reply:
Ha!  That was it!  You just saved me hours of debugging.